flink-runtime/src/test/java/org/apache/flink/runtime/jobmaster/JobMasterTest.java

```diff
@@ -1452,20 +1345,17 @@ public JobGraph createKvJobGraph() {
 	public void testRequestKvStateWithIrrelevantRegistration() throws Exception {
 		final JobGraph graph = createKvJobGraph();
 
-		final JobMaster jobMaster = createJobMaster(
-			configuration,
-			graph,
-			haServices,
-			new TestingJobManagerSharedServicesBuilder().build(),
-			heartbeatServices);
+		final JobMaster jobMaster = new JobMasterBuilder(graph, rpcService)
+			.withConfiguration(configuration)
+			.withHighAvailabilityServices(haServices)
+			.withHeartbeatServices(heartbeatServices)
+			.createJobMaster();
+
+		jobMaster.start();
 
-		CompletableFuture<Acknowledge> startFuture = jobMaster.start(jobMasterId);
 		final JobMasterGateway jobMasterGateway = jobMaster.getSelfGateway(JobMasterGateway.class);
 
 		try {
-			// wait for the start to complete
-			startFuture.get(testingTimeout.toMilliseconds(), TimeUnit.MILLISECONDS);
-
 			// register an irrelevant KvState
 			try {
 				jobMasterGateway.notifyKvStateRegistered(
```

Comment:
How do we guarantee that the JobMaster has already fully started at this point?
Until JobMaster#onStart has finished the AkkaRpcActor is still officially in a stopped state and would reject any incoming message, correct?

Reply:
There is no need for this. After `jobMaster.start()` is being called all subsequent messages which are sent to the `JobMaster` will be executed. The reason is that we only have a single mailbox which is processed one after another: `START; notifyKvStateRegistered; ....`